### PR TITLE
Initial layout for the challenge

### PR DESCRIPTION
**Pull request by lankastersky**

### CODE REVIEW

File: zen/models/ChallengeLevel.swift [hunk 1]
```diff
@@ -1,5 +1,15 @@
+import Foundation
+
 enum ChallengeLevel: Int, Codable {
     case low = 1
     case medium
     case high
+
+    var description: String {
+        switch self {
+        case .low: return NSLocalizedString("low", comment: "")
```
Comment (NikAshanin) [inline zen/models/ChallengeLevel.swift]: create an extension for NSLocalizedString("low", comment: "")

File: zen/en.lproj/Localizable.strings [hunk 2]
```diff
@@ -0,0 +1,11 @@
+"Current Challenge" = "Current Challenge";
```
Comment (NikAshanin) [inline zen/en.lproj/Localizable.strings]: better to create a key with a better naming including flow, screen and element like 
https://github.com/TouchInstinct/team/blob/master/processes/Strings-And-Images-Naming-Rules.md

File: zen/views/ChallengeViewController.swift [hunk 3]
```diff
@@ -7,42 +7,102 @@ final class ChallengeViewController: UIViewController {
...
+        detailsLabel.text = challenge.details
+        quoteLabel.text = challenge.quote
+        typeLabel.text = "\(NSLocalizedString("Type", comment: "")): \(challenge.type)"
+        levelLabel.text =
+        "\(NSLocalizedString("Difficulty", comment: "")): \(challenge.level.description)"
+
+        challengesService.markChallengeShown(challenge.challengeId)
+    }
+
+    @IBAction func onChallengeButton(sender: UIButton!) {
```
Comment (NikAshanin) [inline zen/views/ChallengeViewController.swift]: private

**Review by NikAshanin — CHANGES_REQUESTED** [review]

File: zen/models/ChallengeLevel.swift (hunk 1 above)
Comment (lankastersky) [reply zen/models/ChallengeLevel.swift]: it's a function, not class

File: zen/views/ChallengeViewController.swift (hunk 3 above)
Comment (lankastersky) [reply zen/views/ChallengeViewController.swift]: done

File: zen/models/ChallengeLevel.swift (hunk 1 above)
Comment (NikAshanin) [reply zen/models/ChallengeLevel.swift]: extenstion for String...

File: zen/en.lproj/Localizable.strings (hunk 2 above)
Comment (lankastersky) [reply zen/en.lproj/Localizable.strings]: done

Comment (lankastersky) [reply zen/en.lproj/Localizable.strings]: done

File: zen/models/ChallengeLevel.swift (hunk 1 above)
Comment (lankastersky) [reply zen/models/ChallengeLevel.swift]: it's a function, not class

File: zen/views/JournalViewController.swift [hunk 4]
```diff
@@ -22,6 +19,11 @@ final class JournalViewController: UIViewController {
             flowLayout.estimatedItemSize = CGSize(width: 1, height: 1)
         }
     }
+
+    override func viewWillAppear(_ animated: Bool) {
+        challenges = challengesService.finishedChallengesSortedByTimeDesc()
```
Comment (NikAshanin) [inline zen/views/JournalViewController.swift]: super забыл

File: zen/views/JournalViewController.swift [hunk 5]
```diff
@@ -44,22 +46,56 @@ extension JournalViewController: UICollectionViewDataSource {
...
         let challenge: Challenge = challenges[indexPath.item]
+        cell.finishedTimeLabel.text =
+            Date(timeIntervalSince1970: challenge.finishedTime).toStringWithMediumFormat()
         cell.contentLabel.text = challenge.content
         cell.detailsLabel.text = challenge.details
-        cell.backgroundColor = randomColor()
+        cell.levelLabel.text =
+            "\("current_challenge_screen_difficulty".localized): \(challenge.level.description)"
+        if let rating = challenge.rating {
+            cell.cosmosView.rating = rating
```
Comment (NikAshanin) [inline zen/views/JournalViewController.swift]: используй тернарный оператор

File: zen/views/JournalViewController.swift [hunk 6]
```diff
@@ -44,22 +46,56 @@ extension JournalViewController: UICollectionViewDataSource {
...
+            "Bad challenge index when creating collection view"
+        )
+
+        let challenge: Challenge = challenges[indexPath.item]
+        openChallengeView(challenge)
+    }
+
+    private func openChallengeView(_ challenge: Challenge) {
+        guard let challengeViewController =
+            self.storyboard?.instantiateViewController(
```
Comment (NikAshanin) [inline zen/views/JournalViewController.swift]: self redundant

File: zen/views/JournalCollectionViewCell.swift [hunk 7]
```diff
@@ -19,6 +20,6 @@ final class JournalCollectionViewCell: UICollectionViewCell {
         // TODO: consider using table cells instead.
         contentView.translatesAutoresizingMaskIntoConstraints = false
         let screenWidth = UIScreen.main.bounds.size.width
-        widthConstraint.constant = screenWidth - 2 * 12
+        widthConstraint.constant = screenWidth - 2 * 0
```
Comment (NikAshanin) [inline zen/views/JournalCollectionViewCell.swift]: 2*0?))

**Review by NikAshanin — CHANGES_REQUESTED** [review]

File: zen/views/JournalViewController.swift (hunk 4 above)
Comment (lankastersky) [reply zen/views/JournalViewController.swift]: done

File: zen/views/JournalViewController.swift (hunk 6 above)
Comment (lankastersky) [reply zen/views/JournalViewController.swift]: done

File: zen/views/JournalCollectionViewCell.swift (hunk 7 above)
Comment (lankastersky) [reply zen/views/JournalCollectionViewCell.swift]: removed

File: zen/views/JournalViewController.swift (hunk 5 above)
Comment (lankastersky) [reply zen/views/JournalViewController.swift]: для меня это звучит как заклинание) ок, погуглил применил

**Review by NikAshanin — APPROVED** [review]

**Pull request merged**